plugins/woocommerce-admin/client/marketplace/components/tabs/tabs.tsx

```diff
@@ -67,10 +71,25 @@ const setUrlTabParam = ( tabKey: string ) => {
 	} );
 };
 
-const renderTabs = ( contextValue: MarketplaceContextType ) => {
+const getVisibleTabs = ( selectedTab: string ) => {
+	if ( selectedTab === '' ) {
+		return tabs;
+	}
+	const currentVisibleTabs = { ...tabs };
+	if ( selectedTab !== 'search' ) {
+		delete currentVisibleTabs.search;
+	}
+
+	return currentVisibleTabs;
+};
+
+const renderTabs = (
+	contextValue: MarketplaceContextType,
```

Comment:
Also for consistency, I think we should name this `marketplaceContextValue`, and also on line 90.